iwyu.cc

```diff
@@ -2224,6 +2224,32 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     return true;
   }
 
+  bool VisitTypeTraitExpr(clang::TypeTraitExpr* expr) {
+    if (CanIgnoreCurrentASTNode())  return true;
+
+    // We assume that all type traits with >= 2 arguments require
+    // full type information even for pointer types. For example,
+    // this is the case for `__is_convertible_to` trait.
+    if (expr == nullptr || expr->getNumArgs() <= 1)  return true;
+
+    for (const clang::TypeSourceInfo* arg : expr->getArgs()) {
+      auto qual_type = arg->getType();
+      const Type* type = qual_type.getTypePtr();
+      if (const PointerType* pointer_type =
```

Comment:
Here you _could_ use `const auto*` to shorten line length (as ` PointerType` is clearly visible in the cast expr)